MichiganDaily/parcel-transformer-nunjucks
# Add basename var to nunjucks

#### What's this PR do?

Exposes the basename of the [asset](https://parceljs.org/plugin-system/transformer/#BaseAsset) being transformed to nunjucks so it can be accessed in a template file. In our use, the asset would be an HTML file. Probably something like `config.entries[basename(asset.filePath)]`.

#### Are there any relevant screenshots?

N/A

#### Why are we doing this? How does it help us?

This helps us in <https://github.com/MichiganDaily/cookie/pull/35>

#### How should this be manually tested?

You can test that this works by testing <https://github.com/MichiganDaily/cookie/pull/35>. Alternatively, you can test in this repository by doing the following:

1. Clone the repository and switch to this branch.
2. Change the directory into `test`.
3. Modify the `index.html` file by adding `{{ file }}` somewhere.
4. Run `yarn install` in `test`.
5. Once dependencies install, there should be a `dist` folder with `index.html`.
6. Open `dist/index.html`. You should see `index.html` in place of `{{ file }}`.

#### Are there any smells or added technical debt to note?

Not particularly. But, there could be an alternative solution where we don't pass the filename as a nunjucks variable. Instead, we use the filename just in the transformer to only send the relevant entry in the config.

#### What are relevant issues or links?

<https://github.com/MichiganDaily/cookie/pull/35>

#### Have you done the following, if applicable:
